TACC/Core-Portal
# Task/WP-724: Mutation Hooks: Compress Files

## Overview

We are working on updating our current mutations to utilize TypeScript and React Query instead of React Redux. This should improve the speed and functionality of all portal systems going forward.

Changes the `compressApp` and `extractApp` app parameters to objects:
```
    "compressApp": {
        "id": "compress",
        "version": "0.0.3"  # Can be set to "" to use the latest version
    },
    "extractApp": {
        "id": "extract",
        "version": "0.0.1"  # Can be set to "" to use the latest version
    },
```

## Related

* [WP-724](https://tacc-main.atlassian.net/browse/WP-724)

## Changes

I converted `useCompress.js` to `useCompress.ts` and created a mutation that compresses the selected files into a ZIP file that gets placed in the Root directory.

## Testing

1. Navigate to any folder containing multiple files.
2. Select 2 or more files and click on the Compress button in the Toolbar.
3. Click on the Compress button in the modal that appears.
4. The modal will close, and you get a notification that the ZIP file will be in your Root directory shortly.

## UI

![image](https://github.com/user-attachments/assets/e4d31db4-8cfd-4e9a-a76c-16782932b4a4)

## Notes

1. Folders cannot be compressed.
2. The total size of all files to be compressed must be less than 2 GB.
